model/ftn/ww3_prnc.ftn

```diff
@@ -1809,6 +1812,33 @@
                           + RD21(IX,IY) * XC(IX22(IX,IY),IY21(IX,IY)) &
                           + RD12(IX,IY) * XC(IX21(IX,IY),IY22(IX,IY)) &
                           + RD22(IX,IY) * XC(IX22(IX,IY),IY22(IX,IY))
+
+                    ! Correction for points along the land/sea interface:
+                    ! If one point used in the interpolation has value
+                    ! zero, it should not be used as it is not part of
+                    ! the sea grid, and the rest of the interpolation
+                    ! coefficients scaled so that they sum 1 - otherwise
+                    ! we will create unphysical values along the coast
+                    NORM = 0.0
+                    IF(ABS(XC(IX21(IX,IY),IY21(IX,IY))).GT.EPS)       &
```

Comment:
Is it safe to assume that cells with a value of zero should not be included in the interpolation? There might be legitimate cases where a zero value is valid for a field (such as wind speed, current speed, etc).

Reply:
I agree that a zero value is sometimes a valid value, but it is also the value that is set by default to the land points - and by extension the value of boundary points that are not ocean points. The best way would be using this piece of code only if the grid point we are dealing with is not a boundary point. I will investigate, thanks for the useful comment.

Reply:
>  but it is also the value that is set by default to the land points

What system is setting the land points as zero? WW3? The OASIS coupler? Is it something we have control of as it would be better if it was set to a dedicated "missing value" rather than zero?

Reply:
In one WW3 routine all the values read from the input forcing that contain a fill value are set to zero. I am changing the code to check for MAPSTA instead to see if this works.

Reply:
I have found a better solution but it is more involved - it will take me a while to sort everything out, I may create a routine so that I do not repeat code.

Reply:
> 
> 
> In one WW3 routine all the values read from the input forcing that contain a fill value are set to zero. I am changing the code to check for MAPSTA instead to see if this works.

Ah yes - that is ringing a bell now. I think I ran into this problem with v4.18 of ww3_prnc.